preflight_check/main.py

```diff
@@ -0,0 +1,250 @@
+For each --kill-* testcase, report directory: {}'''.format(context.logfile,
+                                                           context.jsonfile,
+                                                           context.report_path))
+
+    parser.add_argument('-e', '--env-check', dest='env_check', action='store_true',
+                        help='Check environment')
+    parser.add_argument('-c', '--cluster-check', dest='cluster_check', action='store_true',
+                        help='Check cluster state')
+
+    group_mutual = parser.add_mutually_exclusive_group()
+    group_mutual.add_argument('--kill-sbd', dest='sbd', action='store_true',
+                              help='Kill sbd daemon')
+    group_mutual.add_argument('--kill-corosync', dest='corosync', action='store_true',
+                              help='Kill corosync daemon')
+    group_mutual.add_argument('--kill-pacemakerd', dest='pacemakerd', action='store_true',
+                              help='Kill pacemakerd daemon')
+    group_mutual.add_argument('--fence-node', dest='fence_node', metavar='NODE',
+                              help='Fence specific node')
+    group_mutual.add_argument('--split-brain-iptables', dest='sp_iptables', action='store_true',
+                              help='Make split brain by blocking corosync ports')
+    parser.add_argument('-l', '--kill-loop', dest='loop', action='store_true',
+                        help='Kill process in loop')
+
+    other_options = parser.add_argument_group('other options')
+    other_options.add_argument('-y', '--yes', dest='yes', action='store_true',
+                               help='Answer "yes" if asked to run the test')
+    other_options.add_argument('-h', '--help', dest='help', action='store_true',
+                               help='show this help message and exit')
+
+    args = parser.parse_args()
+    if args.help:
+        parser.print_help()
+        sys.exit(0)
+    for arg in vars(args):
+        setattr(context, arg, getattr(args, arg))
+
+
+def setup_logging(context):
+    """
+    Setupt logging
+    """
+    LOGGING_CFG['handlers']['file']['filename'] = context.logfile
+    logging.config.dictConfig(LOGGING_CFG)
+
+
+def setup_basic_context(context):
+    """
+    Setup basic context
+    """
+    var_dir = "/var/lib/{}".format(context.process_name)
+    context.var_dir = var_dir
+    context.report_path = var_dir
+    context.jsonfile = "{}/{}.json".format(var_dir, context.process_name)
+    context.logfile = "/var/log/{}.log".format(context.process_name)
+
+
+def run(context):
+    """
+    Major work flow
+    """
+    setup_basic_context(context)
+    parse_argument(context)
+    if not utils.is_root():
+        logging.fatal("{} can only be executed as user root!".format(context.process_name))
+        sys.exit(1)
+    if not os.path.exists(context.var_dir):
+        os.makedirs(context.var_dir, exist_ok=True)
+    setup_logging(context)
+
+    try:
+        check.check(context)
```

Comment:
Personally, i would perfer to check each flag of `context`, then call the responding function. 
Instead of always call the function with `context`, do nothing if the `flag` not set.

But it is not a bug that must be fixed.

Reply:
The reason I write it in this way was to keep the block size of `run` function short and clear